includes/User.php

```diff
@@ -592,6 +592,40 @@ public static function whoIs( $id ) {
 	}
 
 	/**
+	 *
+	 * @param $ids Array User IDs
+	 * @return Array User ID to User name mapping
+	 */
+	public static function whoAre( Array $ids, $source = DB_SLAVE ): Array {
+		global $wgExternalSharedDB;
+		$sdb = wfGetDB( $source, [], $wgExternalSharedDB );
+		$res = $sdb->select(
+			'`user`',
+			[ 'DISTINCT user_id', 'user_name' ],
+			[ 'user_id' => $ids ],
+			__METHOD__
+		);
+
+		// Pre-fill the returned array with empty strings
+		// so that missing users are not skipped.
+		// It makes further iterating over the array
+		// and handling anons and missing users a little
+		// bit easier.
+		$users = array_fill_keys( $ids, '' );
+
+		// This is optional, let's decide whether it is actually
+		// convenient to handle it here or programmers using
+		// this method should handle it themselves.
+		$users[0] = wfMessage( 'oasis-anon-user' )->escaped();
```

Comment:
I'd rather return `false` to make the behaviour here similar to the of `User::whoIs`:

```
@return String|false The corresponding username
```

Reply:
Consistency is a value, but being consistently wrong?  Considering PHP 7 returning type declarations in function signatures, how would you declare `User::whoIs()`?  User names cannot be empty as in `''`.  That's why it is a sign of a problem when they are.  There is no need to return `false`, or even worse `null`.